tools/shared/src/main/scala/org/scalajs/core/tools/linker/checker/IRChecker.scala

```diff
@@ -1230,23 +1245,13 @@ private final class IRChecker(unit: LinkingUnit,
 
     def fromSignature(thisType: Type, params: List[ParamDef],
         resultType: Type, isConstructor: Boolean = false): Env = {
-      val paramLocalDefs = params.foldLeft(Map.empty[String, LocalDef]) {
-        case (prevParams, p @ ParamDef(ident, tpe, mutable, _)) =>
-          implicit val ctx = ErrorContext(p)
-          val name = ident.name
-          if (prevParams.contains(name))
-            reportDuplicateLocalVarName(name)
-          prevParams + (name -> LocalDef(name, tpe, mutable)(p.pos))
-      }
-      new Env(thisType, paramLocalDefs,
+      val paramLocalDefs =
+        for (p @ ParamDef(ident, tpe, mutable, _) <- params)
+          yield ident.name -> LocalDef(ident.name, tpe, mutable)(p.pos)
```

Comment:
Why didn't you call `checkDeclareLocalVar` in here, instead of adding calls to it in various parts dealing with `ParamDef`s?

Reply:
Because:
https://github.com/sjrd/scala-js/commit/f47c51f76944739b43c1928444e99884b1128738 :)

I can check here as well, do you think it is cleaner overall?

Reply:
Hum ... Yes, I think it's cleaner if it's done once and for all in `fromSignature`, don't you think?

Reply:
Yeah, probably you're right... I'll change it.

Reply:
Updated. I tried doing this and really did not like it, notably because we only create `Env`'s lazily when we traverse into the trees. So in some cases, parameter checking would not happen, even if it could. IMO it is cleaner overall if we keep the `Env` side effect free.

Reply:
Ah I see. Let's keep it, then.